runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -209,9 +214,15 @@ public long TXEnd() throws TransactionAbortedException {
      */
     public void gc(long trimMark) {
         for (Object obj : getObjectCache().values()) {
-            ((CorfuCompileProxy) ((ICorfuSMR) obj).
-                    getCorfuSMRProxy()).getUnderlyingObject().gc(trimMark);
+            if (((ICorfuSMR) obj).getCorfuSMRProxy() instanceof CorfuCompileProxy) {
+                ((CorfuCompileProxy) ((ICorfuSMR) obj).
+                        getCorfuSMRProxy()).getUnderlyingObject().gc(trimMark);
+            }
         }
+
+        mvoCache.getAllMVOs().forEach((uuid, mvo) -> {
+            ((MultiVersionObject) mvo).gc(trimMark);
+        });
```

Comment:
Ok, I see how both types are being gc'd. Maybe introduce a generic gc method in `ICorfuSMRProxy?`